src/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/DirichletFreeOutflow.cpp

```diff
@@ -8,18 +8,269 @@
+    const Scalar<DataVector>& interior_lorentz_factor,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& interior_spatial_velocity,
+    const tnsr::I<DataVector, 3, Frame::Inertial>& interior_magnetic_field,
+
+    // fd_gridless_tags
+    const double time,
+    const std::unordered_map<
+        std::string,
+        std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time,
+    const ElementMap<3, Frame::Grid>& logical_to_grid_map,
+    const domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, 3>&
+        grid_to_inertial_map,
+    const fd::Reconstructor& reconstructor) const {
+  const size_t ghost_zone_size{reconstructor.ghost_zone_size()};
+
+  const auto ghost_logical_coords =
+      evolution::dg::subcell::fd::ghost_zone_logical_coordinates(
+          subcell_mesh, ghost_zone_size, direction);
+
+  const auto ghost_inertial_coords = grid_to_inertial_map(
+      logical_to_grid_map(ghost_logical_coords), time, functions_of_time);
+
+  // Compute FD ghost data with the analytic data or solution
+  auto boundary_values = call_with_dynamic_type<
+      tuples::TaggedTuple<gr::Tags::SpacetimeMetric<DataVector, 3>,
+                          ::gh::Tags::Pi<DataVector, 3>,
+                          ::gh::Tags::Phi<DataVector, 3>>,
+      ghmhd::GhValenciaDivClean::InitialData::analytic_solutions_and_data_list>(
+      analytic_prescription_.get(),
+      [&ghost_inertial_coords, &time](const auto* const initial_data) {
+        using spacetime_tags =
+            tmpl::list<gr::Tags::SpacetimeMetric<DataVector, 3>,
+                       ::gh::Tags::Pi<DataVector, 3>,
+                       ::gh::Tags::Phi<DataVector, 3>>;
+        if constexpr (is_analytic_solution_v<
+                          std::decay_t<decltype(*initial_data)>>) {
+          return initial_data->variables(ghost_inertial_coords, time,
+                                         spacetime_tags{});
+        } else if constexpr (evolution::is_numeric_initial_data_v<
+                                 std::decay_t<decltype(*initial_data)>>) {
+          ERROR(
+              "Cannot currently use numeric initial data as an analytic "
+              "prescription for boundary conditions.");
+        } else {
+          (void)time;
+          return initial_data->variables(ghost_inertial_coords,
+                                         spacetime_tags{});
+        }
+      });
+
+  *spacetime_metric =
+      get<gr::Tags::SpacetimeMetric<DataVector, 3>>(boundary_values);
+  *pi = get<::gh::Tags::Pi<DataVector, 3>>(boundary_values);
+  *phi = get<::gh::Tags::Phi<DataVector, 3>>(boundary_values);
+
+  // Note: Once we support high-order fluxes with GHMHD we will need to
+  // handle this correctly.
+  std::optional<Variables<db::wrap_tags_in<
+      Flux, typename grmhd::ValenciaDivClean::System::flux_variables>>>
+      cell_centered_ghost_fluxes{std::nullopt};
+  // Set to zero since it shouldn't be used
```

Comment:
I think this comment can be made more clear. e.g. dummy placeholders to call `fd_ghost_impl` but not actually returned.